uniswap/decorators.py

```diff
@@ -20,28 +21,30 @@
 P = ParamSpec("P")
 
 
-def check_approval(
-    method: Callable[Concatenate["Uniswap", P], T]
-) -> Callable[Concatenate["Uniswap", P], T]:
-    """Decorator to check if user is approved for a token. It approves them if they
+def check_approval(method: Callable[..., T]) -> Callable[..., T]:
+    """Decorator to check if the user is approved for a token. It approves them if they
     need to be approved."""
 
     @functools.wraps(method)
-    def approved(self: "Uniswap", *args: P.args, **kwargs: P.kwargs) -> T:
+    def approved(self: "Uniswap", *args: Union[AddressLike, int], **kwargs: AddressLike) -> T:
```

Comment:
This ruins typing (and is not even valid, `*args` has to be an iterable and `**kwargs` a mapping) for any function annotated with `check_approval`. The `ParamSpec` is used for a reason.